tests/CacheTest.php

```diff
@@ -20,92 +22,131 @@ public function setUp()
+        $this->registrar->getPermissions();
 
-        DB::flushQueryLog();
+        $this->assertQueryCount(self::QUERIES_PER_CACHE_PROVISION);
     }
 
     /** @test */
-    public function it_can_cache_the_permissions()
+    public function it_flushes_the_cache_when_creating_a_permission()
     {
-        $this->registrar->registerPermissions();
+        app(Permission::class)->create(['name' => 'new']);
+
+        $this->resetQueryCount();
 
-        $this->assertCount(0, DB::getQueryLog());
+        $this->registrar->getPermissions();
+
+        $this->assertQueryCount(self::QUERIES_PER_CACHE_PROVISION);
     }
 
     /** @test */
-    public function permission_creation_and_updating_should_flush_the_cache()
+    public function it_flushes_the_cache_when_updating_a_permission()
     {
         $permission = app(Permission::class)->create(['name' => 'new']);
-        $this->assertCount(1, DB::getQueryLog());
 
-        $this->registrar->registerPermissions();
-        $this->assertCount(3, DB::getQueryLog());
+        $this->resetQueryCount();
 
         $permission->name = 'other name';
         $permission->save();
-        $this->assertCount(4, DB::getQueryLog());
 
-        $this->registrar->registerPermissions();
-        $this->assertCount(6, DB::getQueryLog());
+        $this->resetQueryCount();
+
+        $this->registrar->getPermissions();
+
+        $this->assertQueryCount(self::QUERIES_PER_CACHE_PROVISION);
     }
 
     /** @test */
-    public function role_creation_and_updating_should_flush_the_cache()
+    public function it_flushes_the_cache_when_creating_a_role()
     {
-        $role = app(Role::class)->create(['name' => 'new']);
-        $this->assertCount(2, DB::getQueryLog());
+        app(Role::class)->create(['name' => 'new']);
+
+        $this->resetQueryCount();
 
-        $this->registrar->registerPermissions();
-        $this->assertCount(4, DB::getQueryLog());
+        $this->registrar->getPermissions();
+
+        $this->assertQueryCount(self::QUERIES_PER_CACHE_PROVISION);
+    }
+
+    /** @test */
+    public function it_flushes_the_cache_when_updating_a_role()
+    {
+        $role = app(Role::class)->create(['name' => 'new']);
 
         $role->name = 'other name';
         $role->save();
-        $this->assertCount(5, DB::getQueryLog());
 
-        $this->registrar->registerPermissions();
-        $this->assertCount(7, DB::getQueryLog());
+        $this->resetQueryCount();
+
+        $this->registrar->getPermissions();
+
+        $this->assertQueryCount(self::QUERIES_PER_CACHE_PROVISION);
     }
 
     /** @test */
     public function user_creation_should_not_flush_the_cache()
     {
+        $this->registrar->getPermissions();
+
         User::create(['email' => 'new']);
-        $this->assertCount(1, DB::getQueryLog());
 
-        $this->registrar->registerPermissions();
-        $this->assertCount(1, DB::getQueryLog());
+        $this->resetQueryCount(1);
```

Comment:
`resetQueryCount()` doesn't take any parameters, so the `1` here is superfluous.